components/org.wso2.carbon.identity.auth.abstractotp/src/main/java/org/wso2/carbon/identity/auth/abstractotp/internal/AuthenticatorDataHolder.java

```diff
@@ -0,0 +1,142 @@
+package org.wso2.carbon.identity.auth.abstractotp.internal;
+
+import org.wso2.carbon.identity.event.services.IdentityEventService;
+import org.wso2.carbon.identity.governance.IdentityGovernanceService;
+import org.wso2.carbon.identity.handler.event.account.lock.service.AccountLockService;
+import org.wso2.carbon.idp.mgt.IdpManager;
+import org.wso2.carbon.user.core.service.RealmService;
+
+/**
+ * Encapsulates the data of OTP authenticator.
+ */
+public class AuthenticatorDataHolder {
+
+    private static volatile AuthenticatorDataHolder authenticatorDataHolder = new AuthenticatorDataHolder();
```

Comment:
Why do you need a volatile variabale here?

Reply:
Removed volatile keyword